tests/functional/cylc-poll/04-poll-multi-hosts/suite.rc

```diff
@@ -5,46 +5,44 @@
         expected task failures = local-fail-1.1, local-fail-2.1, remote-fail-1.1
 [scheduling]
     [[graph]]
-        R1="""
-POLLABLE:start-all => poller
-"""
+        R1 = """
+            POLLABLE:start-all => poller
+        """
 [runtime]
     [[POLLABLE]]
-        pre-script="""
-# Stop script from reporting anything back
-trap '' 'EXIT'
-trap '' 'ERR'
-"""
+        pre-script = """
+            # Stop script from reporting anything back
+            trap '' 'EXIT'
+            trap '' 'ERR'
+        """
```

Comment:
I didn't previously have any awareness of strong opinions on this matter. Thanks.

Reply:
This was most-likely me. If I can't read a suite.rc quickly I tend to "fix" the indentation.

> (graph strings and script items) should not be indented as if it was part of the surrounding suite.rc syntax

*(Ok, here's the start of a long rant which Tim's PR should really be spared...)*

**[rant]**

I don't think I could possibly disagree more! Multi-line strings *are* a part of the suite.rc syntax why on earth would we want to treat them as alien to the file format?

Multi-line string *should always* (unless necessary for bash syntax) be indented as if they are part of the surrounding `suite.rc` for:

* Readability & clarity.
* To help prevent indentation/suiterc-nesting errors (e.g. `[scheduling][graph][xtriggers]`) which are soooo common.
* Code folding in text editors.

This is pretty bad from the perspective of readability.

```ini
# The visual association between `bar` and `a` is stronger
# than the association between `a` and `b`
[foo]
    bar = """a
        b"""
[baz]
```

Especially when the strings get long they make a complete mess of the file flow making it very hard to work out what section we are inside later on in the file.

This approach with the trailing `"""` at a different indentation to the setting is bad for readability
but is also potentially miss-leading from an indentation perspective:

```ini
[foo]
    bar = """a
        b
        """
        # ^ this """ makes it look like we are in a second order section
        # whereas we are actually only in a first order section
[baz]
```

Moreover what logic is there for putting the closure at a different indentation level, this doesn't happen in other languages...

```bash
if true; then
    echo success
    fi
    # this is clearly misleading
```

No indentation doesn't parse well either:

```ini
# now `a` and `b` are visually associated and the closing """ is at the right level
# so the indentation/nesting-level are implcit, however, the flow of the file has
# been broken making it hard to scan and breaking code-folding
[foo]
    bar = """
a
b
    """
    # it is clear that we are in a first-level section even if [foo]
    # is not visible because the indentation makes this implicit
[baz]
```

But if we work with the indentation of the suite.rc rather than fighting it things work much better:

```ini
[foo]
    bar = """
        a
        b
    """
[baz]
```

The visual associations are all correct `foo>bar>a`, the setting closes at the same level at which is was opened which makes the code-flow clear, the indentation/nesting-level is implicit, code-folding works, you can comment out lines in the graph string without breaking the `suite.rc` syntax.

Closing notes:

*YAML*

```yaml
foo:
   bar: |
      a
      b
```

*HTML*

```html
<foo>
  <bar>
    a
    b
  </bar>
</foo>
```

*Python*

```python
class Foo:
    def bar:
        """
            a
            b
        """

```

**[/rant]**

Reply:
He heh :rofl:  Well I've already accepted defeat on this long ago, and luckily it makes no difference in practical terms for our current strings so it's a moot point.  But **[unassailable-logic]** as a matter of principle you're clearly dead wrong. These multi-line strings are config item values and the validity of their content obviously depends on exactly how the values re used, which has absolutely nothing to do with the style conventions of the surrounding config file. If we had Python job scripts, for instance, we'd have to be much more careful with indentation inside the `script` strings.  The fact that we can alter the string content after parsing and before use, to avoid such problems, does not negate my point - that is just a workaround, which could potentially be fragile. **[/unassailable-logic]**

Reply:
Python doc strings are not a good comparison, BTW - they are by definition arbitrary (to Python) comments (and any 3rd party applications that use them have to be aware of that).  Our `script` strings are verbatim bits of an external bash script (the task job script) and therefore have to conform to the syntax (and other) rules of bash, not Cylc.

Reply:
Readability is important, of course, but it is (somewhat) subjective and it doesn't trump syntactic correctness.  (I might argue, for instance, that all Python class method definitions should start at the left margin for readability - it makes them stand out more! - but I'm not allowed to do that).